### PR TITLE
(SERVER-2993) Detect migrated CA dir when calcuating settings

### PR DESCRIPTION
**Pull request by Magisus**
This commit updates the settings resolution logic to detect when the CA
dir has been migrated, to return the new location in that case. If the
new dir does not have CA content, the old location is returned
(including if no dir exists).

### CODE REVIEW

**Comment by Magisus** [comment]
fwiw in my manual testing, I also verified that this works fine for most operations even without this change (the symlink is sufficient). So if this doesn't make the release, that's fine IMO.

File: spec/puppetserver/ca/config/puppet_spec.rb [hunk 1]
```diff
@@ -69,7 +71,7 @@
       conf.load
 
       expect(conf.errors).to be_empty
-      expect(conf.settings[:cacert]).to eq('/foo/bar/ca/ca_crt.pem')
+      expect(conf.settings[:localcacert]).to eq('/foo/bar/certs/ca.pem')
```
Comment (mwaggett) [inline spec/puppetserver/ca/config/puppet_spec.rb:74]: What's the reason for this change?

File: spec/puppetserver/ca/config/puppet_spec.rb [hunk 2]
```diff
@@ -241,4 +243,73 @@
...
+
+        expect(File).to receive(:exist?)
+          .with("#{File.dirname(confdir)}/puppetserver/ca/ca_crt.pem")
+          .and_return(true)
+
+        conf = Puppetserver::Ca::Config::Puppet.new(puppet_conf)
+        settings = conf.load
+
+        expect(settings[:ssldir]).to eq(File.join(confdir, 'ssl'))
+        expect(settings[:cadir]).to eq(File.join(File.dirname(confdir), 'puppetserver/ca'))
```
Comment (mwaggett) [inline spec/puppetserver/ca/config/puppet_spec.rb:290]: might be worth putting `File.dirname(confdir)` into a variable, since it's used three times ¯\_(ツ)_/¯ I don't feel super strongly though.

**Review by mwaggett — APPROVED** [review]

**Comment by mwaggett** [comment]
I'm guessing those Travis failures are still the Facter thing, but I really don't love having to just ignore test failures. We don't have to address it here, but it seems valuable to have a way to easily switch to dev versions of facter for testing purposes.

File: spec/puppetserver/ca/config/puppet_spec.rb (hunk 1 above)
Comment (Magisus) [reply spec/puppetserver/ca/config/puppet_spec.rb:74]: Justin also made this change in his original PR for this on the other branch: the problem is that the location of `cacrt` becomes unpredictable with this change, without a bunch of mocking, but this is testing verifying that we pick up the customer `ssldir` location. So this change switches to testing a setting that still interpolates that.

**Comment by Magisus** [comment]
We might be able to override it in the gemfile to point at Git? Fwiw these tests only fail on Travis, they pass for me locally.

File: spec/puppetserver/ca/config/puppet_spec.rb (hunk 1 above)
Comment (mwaggett) [reply spec/puppetserver/ca/config/puppet_spec.rb:74]: ahh gotcha, thanks

**Review by mwaggett — APPROVED** [review]
